### PR TITLE
[Tables] mypy

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

**Comment by seankane-msft** [comment]
closes #18135

File: sdk/tables/azure-data-tables/azure/data/tables/_table_client.py [hunk 1]
```diff
@@ -210,13 +213,13 @@ def set_table_access_policy(
...
             self._client.table.set_access_policy(
-                table=self.table_name, table_acl=signed_identifiers or None, **kwargs
+                table=self.table_name, table_acl=signed_identifiers or None, **kwargs  # type: ignore
             )
         except HttpResponseError as error:
             try:
                 _process_table_error(error)
             except HttpResponseError as table_error:
-                if (table_error.error_code == 'InvalidXmlDocument'
+                if (table_error.error_code == 'InvalidXmlDocument'  # type: ignore
```
Comment (kristapratico) [inline sdk/tables/azure-data-tables/azure/data/tables/_table_client.py:222]: Does HttpResponseError have an error_code on it? I think it was accessed as table_error.error.code

File: sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py [hunk 2]
```diff
@@ -204,25 +204,25 @@ async def set_table_access_policy(
                 value.start = serialize_iso(value.start)
                 value.expiry = serialize_iso(value.expiry)
             identifiers.append(SignedIdentifier(id=key, access_policy=value))
-        signed_identifiers = identifiers  # type: ignore
+        # signed_identifiers = identifiers  # type: ignore
```
Comment (kristapratico) [inline sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py:207]: can we just delete this?

**Review by kristapratico — APPROVED** [review]

File: sdk/tables/azure-data-tables/azure/data/tables/_table_client.py (hunk 1 above)
Comment (seankane-msft) [reply sdk/tables/azure-data-tables/azure/data/tables/_table_client.py:222]: Tables has some custom error processing that adds this property.

**Pull request merged**